### PR TITLE
Add category keys to Parameters

### PR DESCRIPTION
**Pull request by annetill**
**Please check if the PR fulfills these requirements**

- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

No.

**What kind of change does this PR introduce?**

Some category keys are added:
- VoltageControls, GeneratorVoltageControl, TransformerVoltageControl, ShuntVoltageControl.
- ReactivePowerControl
- PhaseControl
- SlackDistribution
- ReferenceBus
- LowImpedanceBranch.
- NewtonRaphson, NewtonKrylov.
- FastRestart
- OuterLoops
- Dc.

Some keys are missing for automaton systems, secondary voltage control, AC solver type, load model, line per unit mode, asymmetrical calculations, extra reported features, debug directory, min and max realistic voltages, voltage init override

**What is the current behavior?**

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [ ] No

**If yes, please check if the following requirements are fulfilled**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration steps are described in the following section

**What changes might users need to make in their application due to this PR? (migration steps)**

**Other information**:

### CODE REVIEW

File: src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java [hunk 1]
```diff
@@ -267,75 +267,98 @@ public static <E extends Enum<E>> List<Object> getEnumPossibleValues(Class<E> en
...
+        new Parameter(MAX_VOLTAGE_CHANGE_STATE_VECTOR_SCALING_MAX_DPHI_PARAM_NAME, ParameterType.DOUBLE, "Max voltage angle change for the max voltage change state vector scaling", MaxVoltageChangeStateVectorScaling.DEFAULT_MAX_DPHI, ParameterScope.TECHNICAL, NEWTON_RAPHSON_CATEGORY_KEY),
         new Parameter(LINE_PER_UNIT_MODE_PARAM_NAME, ParameterType.STRING, "Line per unit mode", LinePerUnitMode.IMPEDANCE.name(), getEnumPossibleValues(LinePerUnitMode.class)),
         new Parameter(USE_LOAD_MODEL_PARAM_NAME, ParameterType.BOOLEAN, "Use load model (with voltage dependency) for simulation", LfNetworkParameters.USE_LOAD_MODE_DEFAULT_VALUE),
-        new Parameter(DC_APPROXIMATION_TYPE_PARAM_NAME, ParameterType.STRING, "DC approximation type", DcEquationSystemCreationParameters.DC_APPROXIMATION_TYPE_DEFAULT_VALUE.name(), getEnumPossibleValues(DcApproximationType.class)),
+        new Parameter(DC_APPROXIMATION_TYPE_PARAM_NAME, ParameterType.STRING, "DC approximation type", DcEquationSystemCreationParameters.DC_APPROXIMATION_TYPE_DEFAULT_VALUE.name(), getEnumPossibleValues(DcApproximationType.class), ParameterScope.FUNCTIONAL, "Dc"),
```
Comment (geofjamg) [inline src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java:351]: We should create a constant for all categories even with one parameter

Comment (annetill) [reply src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java:351]: Yes I will do that. Is the rest of the PR okay for you?

Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java:351]: ideally, we would sort and show the category in the ASCII table of parameters. But that would required for base parameter to also assign in the code a category (here we just have specific parameters one)

Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java:351]: but we can dot it later

Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java]: Done

**Review by geofjamg — APPROVED** [review]

**Comment by jeandemanged** [comment]
very nice! some suggestions:
* `dc` move from category `DC` to `Solver`: then we can see in one place if it is DC, NR, NK
* `PerUnit` category: I would remove this category and merge it in `Model` (only `linePerUnitMode`)
* `reportedFeatures`, I would create a new `Reporting` category, this is not debug.
* Maybe move the `NewtonRaphson` params related to state vector scaling in a dedicated category `NewtonRaphsonStateVectorScaling` (5 parameters), all other parameters remaining in `NewtonRaphson` would then be related to stopping criteria.

**Comment by geofjamg** [comment]
> very nice! some suggestions:
> 
> * `dc` move from category `DC` to `Solver`: then we can see in one place if it is DC, NR, NK
> * `PerUnit` category: I would remove this category and merge it in `Model` (only `linePerUnitMode`)
> * `reportedFeatures`, I would create a new `Reporting` category, this is not debug.
> * Maybe move the `NewtonRaphson` params related to state vector scaling in a dedicated category `NewtonRaphsonStateVectorScaling` (5 parameters), all other parameters remaining in `NewtonRaphson` would then be related to stopping criteria.

Done for perunit et reportedFeatures.
 - DC: DC mode is not only a solver change but above all a modelling change with different equations. This is different that changing a solver like NR, FDNR, NK, GS running on same AC equations.
 -  NewtonRaphsonStateVectorScaling that would required to make it clean to have some kind of subcategory feature in parameters.

**Comment by annetill** [comment]
Looks good!

**Review by jeandemanged — APPROVED** [review]

**Pull request merged**